torch/csrc/jit/script/compiler.cpp

```diff
@@ -46,9 +46,29 @@ struct PrintValue : public SugaredValue {
       auto& g = *m.graph();
       if (!attributes.empty())
         throw ErrorReport(loc) << "print doesn't accept any keyword arguments";
-      auto values = toValues(inputs);
-      ensureTensors(loc, values);
-      g.insertNode(g.create(prim::Print, values, 0)
+
+      //temporary hack to allow print statements to work in python 2, where
+      //print(a, b) is treated as a (a, b) tuple input.
+
+      //logic from lower_tuples, here we are lowering tuples before they are
+      //inputs to the node
+      std::vector<Value*> lowered_inputs = toValues(inputs);
+      for(size_t i = 0; i < lowered_inputs.size();) {
+        auto input = lowered_inputs[i];
+        if(TupleTypePtr tt = input->type()->cast<TupleType>()) {
+          JIT_ASSERTM(input->node()->kind() == prim::TupleConstruct, "tuple use not matched to tuple construct");
```

Comment:
There are a couple problems here.

1. This should only apply if there is a _single_ tuple argument to print. Otherwise you are going to introduce all sorts of weird behavior if people `print(a, (b,c))` for instance.
2. You should not assert that there was a TupleConstruct node. It is perfectly valid for someone to write `print a` where `a` is a tuple argument not produced from a TupleConstruct. Instead this should apply _only_ when there is a single literal tuple construct argument in print:

```
if(lowered_inputs.size() == 1 && lower_inputs.at(0)->node()->kind() == prim::TupleConstruct) {
...
}
```

Reply:
In the case of print(a, (b, c)) this would throw an error if we don't lower all of the tuples. As a temporary measure I was thinking it would be better to lower all tuples, instead of throwing an error

Reply:
Also if we don't lower all tuples wouldn't we have different behavior in python 2 vs 3 in the case of print((a, b))